### PR TITLE
New specification for NIP-51 to accept curated articles (edited from NIP-24 proposition for topics)

### PR DESCRIPTION
**Pull request by mostafa-khaldi**
A NIP that represents topics that group a selection of articles of a `kind:30023` of the NIP-23.

### CODE REVIEW

**Comment by fiatjaf** [comment]
I think this can be just an entry added to NIP-51.

**Comment by mostafa-khaldi** [comment]
Thanks for replying, but what about the data of the topic itself, like the title, the excerpt and the thumbnail, those represent a brief about the topic in which holds the articles.

**Comment by mostafa-khaldi** [comment]
This video demonstrates precisely the case
https://www.youtube.com/watch?v=i5aDU_RLhsM&ab_channel=UNesYu

**Comment by fiatjaf** [comment]
That video reminds me of other Nostr apps that are being worked on that have the same theme (https://listr-demo.vercel.app, https://highlighter.com, https://listr.lol).

@sepehr-safari @pablof7z @erskingardner can we agree on a standard for these lists of curated content?

**Comment by verbiricha** [comment]
>I feel like all this could be shoved into NIP-51 for simplicity and brevity, but I don't know.

+1, also make them tags like @erskingardner suggests.

**Comment by mostafa-khaldi** [comment]
@erskingardner In the specification of the NIP-51 it says that the list should be assigned only once per user, and in our case, multiple topics can be created from one user, therefore this NIP is not fully suitable for topics.
Also for the tag question, are relays accustomed only for the `p` and `d` tags, or do they accept any custom tags from this NIP?

**Comment by pablof7z** [comment]
> @erskingardner In the specification of the NIP-51 it says that the list should be assigned only once per user, and in our case, multiple topics can be created from one user, therefore this NIP is not fully suitable for topics. Also for the tag question, are relays accustomed only for the `p` and `d` tags, or do they accept any custom tags from this NIP?

they can take any tag, one-letter tags are indexed so you can query for them

**Comment by mostafa-khaldi** [comment]
@pablof7z What about the first rule?

**Comment by pablof7z** [comment]
yeah, this seems like a bad idea; these are just NIP-51s. Highlighter.com is already doing this and putting the metadata on tags.

**Comment by erskingardner** [comment]
@mostafa-khaldi I'm not sure what rule you're talking about – a PRE (parameterized replaceable event) can have a name via the "d" tag so a user can create as many lists as they want, giving them whatever name they want. They only issue would be if the user wanted to have multiple lists with the exact same name but different content. Is that what you're referring to?

**Comment by mostafa-khaldi** [comment]
The `d` here is just a reference to the event, but since the relays accept any custom tags, the name of the topic can be embedded in the tags under the `title` tag without conflicting with the `d` reference.

**Comment by fiatjaf** [comment]
@mostafa-khaldi we can adjust the wording of NIP-51. The goal here is to ensure that lists created on one app show up on the other apps. And the `d` tag is specified on NIP-33 in such a way that is useful for these lists.

File: 24.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# NIP-24
+## Topics
+
+This NIP defines `kind:31000` (a parameterized replaceable event according to [NIP-33](https://github.com/nostr-protocol/nips/blob/master/33.md)) for topics that group a selection of articles of a `kind:30023` of the NIP-23.
```
Comment (Giszmo) [inline 24.md:4]:
```suggestion
This NIP defines `kind:31000` (a parametrized replaceable event according to [NIP-33](33.md)) for topics that group a selection of articles of a `kind:30023` of the NIP-23.
```

File: 24.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Format
+The `.content` property should hold a `String` of a `JSON` syntax, a `Stringyfied` JSON.
```
Comment (Giszmo) [inline 24.md:7]:
```suggestion
The `.content` property may contain a stringified `JSON` object.
```

File: 24.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+## Linking
+The article may be linked to using the NIP-19 naddr code along with the "a" tag (see [NIP-33](https://github.com/nostr-protocol/nips/blob/master/33.md) and [NIP-19](https://github.com/nostr-protocol/nips/blob/master/19.md)).
+
+## Example event
+
+```
+{
+  "kind": 31000,
+  "created_at": 1675642635,
+  "content": "{title: "Lorem title", excerpt: "lorem excerpt", thumbnail: "https://example.com/image.png"}",
```
Comment (Giszmo) [inline 24.md:26]:
```suggestion
  "content": "{title: \"Lorem title\", excerpt: \"lorem excerpt\", thumbnail: \"https://example.com/image.png\"}",
```

File: 24.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# NIP-24
+## Topics
+
+This NIP defines `kind:31000` (a parameterized replaceable event according to [NIP-33](https://github.com/nostr-protocol/nips/blob/master/33.md)) for topics that group a selection of articles of a `kind:30023` of the NIP-23.
+
+## Format
+The `.content` property should hold a `String` of a `JSON` syntax, a `Stringyfied` JSON.
+
+The detailed structure of the `Strigyfied` content should hold the following properties:
```
Comment (Giszmo) [inline 24.md:9]:
```suggestion
The strigified content should hold the following properties:
```

File: 24.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+- `title`
+- `excerpt`
+- `thumbnail`
+
+## Editability
+These articles are meant to be editable, so they should make use of the replaceability feature of NIP-33 and include a `d`  tag with an identifier for the article. Clients should take care to only publish and read these events from relays that implement that. If they don't do that they should also take care to hide old versions of the same article they may receive.
+
```
Comment (Giszmo) [inline 24.md:16]: This paragraph is not adding anything substantial. nip-33 already states that clients should not assume relays to support replacement and therefore take the last edit as **the** version. Even if the relay supports it, the client **will always** receive multiple versions whenever the article is being edited.

The fact that this event is meant to be replaceable is already stated in the very first sentence.

```suggestion
```

File: 24.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+## Linking
+The article may be linked to using the NIP-19 naddr code along with the "a" tag (see [NIP-33](https://github.com/nostr-protocol/nips/blob/master/33.md) and [NIP-19](https://github.com/nostr-protocol/nips/blob/master/19.md)).
```
Comment (Giszmo) [inline 24.md:18]:
```suggestion
The article may be linked to using the NIP-19 naddr code along with the "a" tag (see [NIP-33](33.md) and [NIP-19](19.md)).
```

**Comment by sepehr-safari** [comment]
> That video reminds me of other Nostr apps that are being worked on that have the same theme (https://listr-demo.vercel.app, https://highlighter.com, https://listr.lol).
> 
> @sepehr-safari @pablof7z @erskingardner can we agree on a standard for these lists of curated content?
> 
> I feel like all this could be shoved into NIP-51 for simplicity and brevity, but I don't know. @monlovesmango @Semisol @Giszmo @verbiricha @dergigi what do you say?

yeap, I'm currently developing some stuff for managing lists. I've achieved a lot of things so far. I will let you all know about it so we can discuss about a standard for it.

**Comment by mostafa-khaldi** [comment]
@fiatjaf Thank you!

File: 24.md [hunk 7]
```diff
@@ -1,29 +1,26 @@
...
 ## Topics
 
-This NIP defines `kind:31000` (a parameterized replaceable event according to [NIP-33](https://github.com/nostr-protocol/nips/blob/master/33.md)) for topics that group a selection of articles of a `kind:30023` of the NIP-23.
+This NIP defines `kind:31000` (a parametrized replaceable event according to [NIP-33](33.md)) for topics that group a selection of articles of a `kind:30023` of the NIP-23.
 
 ## Format
 The `.content` property should hold a `String` of a `JSON` syntax, a `Stringyfied` JSON.
 
-The detailed structure of the `Strigyfied` content should hold the following properties:
+The strigified content should hold the following properties:
```
Comment (Giszmo) [inline 24.md:9]: I copied your typo :facepalm: 

```suggestion
The stringified content should hold the following properties:
```

**Comment by pablof7z** [comment]
I might be missing something, but why is this a separate NIP, and particularly, a special `kind`? this is just a list, NIP-51 already defines this.

What NIP-51 is missing is a category for lists, but they don't need to be of a special `kind`, this could be as simple as querying with `{"kinds":[30001], "#c:["Articles"]}`

I think this NIP should be scrapped and a `c` tag for categorizing lists should be introduced into NIP-51 separate of the `d` tag.

NIP-51 makes reference to the category as being on the `d` tag but that's insufficient if you consider, e.g. zapstr, where a user might have multiple playlists of the same category (e.g. `"tags": [ ["d", "My main hip-hop playlist"], ["c", "Playlist"] ]`)

**Comment by mostafa-khaldi** [comment]
@pablof7z The idea in making this is an independent NIP is being the category/topic having its own metadata, referring to an entire topic with one word is not enough, it should have its title, excerpt and a thumbnail, and it might carry more data as per the client needs.
The metadata might be stored as tags as you proposed before, `[["title"],["Lorem ipsum dolor sit, amet.."]]` ..etc.

**Comment by pablof7z** [comment]
just store them as tags within a NIP-51 list; you don't need a separate kind or a separate NIP for this.

I am literally doing the same thing with lists descriptions on Atlas Notes; add a `["description", <description-text>]` tag and any other metadata you need as a tag.

**Comment by mostafa-khaldi** [comment]
@pablof7z Yeh that's the thing.

**Comment by mostafa-khaldi** [comment]
@fiatjaf @pablof7z I have added some specifications to NIP-51 in the `kind:30001` to support curated articles.

**Comment by mostafa-khaldi** [comment]
@pablof7z I did no get your point? I think you have missed my new committed changes in the NIP-51, I added new specifications to that NIP using the `kind:30001` with the `c` tag for events that represents curated articles, kindly check the new changes in the file 51.md

**Comment by pablof7z** [comment]
There's still a NIP-24 file that should be axed imo

**Comment by mostafa-khaldi** [comment]
@pablof7z yeh sure, and how about the new specifications in the NIP-51?

**Comment by fiatjaf** [comment]
I think this should be compatible with @sepehr-safari's https://pinstr.app/ and vice-versa.

File: 51.md [hunk 8]
```diff
@@ -110,3 +110,60 @@ An event with kind `30000` is defined as a parameterized replaceable list event
 ### Categorized Bookmarks List
 
 An event with kind `30001` is defined as a parameterized replaceable list event for categorizing bookmarks. The 'd' parameter for this event holds the category name of the list. Any standardized tag can be included in a Categorized Bookmarks List.
+
+#### Curated articles (topics)
```
Comment (monlovesmango) [inline 51.md:114]: #### Categorized Articles List

An event with kind `30002` is defined as a parameterized replaceable list event for categorizing articles. The 'd' parameter for this event holds the category name of the list. The tags included in these lists MUST follow the format of 'a' tags for parameterized replaceable events as defined in [NIP-33 - Parameterized Replaceable Events](33.md#referencing-and-tagging).

Additional tags that can be included as metadata of these lists are:
- `title` 
- `description` 
- `thumbnail` 

Example:
```json
{
    "kind": 30002,
    "pubkey": "8f014c5aafb425b2fa414a49eb9ab17269f155937db67a5babe7bc40f9bbae3b",
    "created_at": 1685011028,
    "content": "",
    "tags": [
        [
            "d",
            "Bitcoin Cultural Movement - English"
        ],
        [
            "title",
            "Bitcoin Cultural Movement"
        ],
        [
            "description",
            "The advent of Bitcoin has not only revolutionized the financial landscape but has also given rise to a new cultural movement centered around this groundbreaking cryptocurrency. In this era of digital innovation, the concept of curation has emerged as a powerful tool to navigate and make sense of the vast amount of information and ideas surrounding Bitcoin. This article delves into the topic of Bitcoin curation, exploring its purpose, key players involved, and presenting five curated articles that shed light on different aspects of this phenomenon."
        ],
        [
            "thumbnail",
            "https://daorayaki-fs-bucket.s3.ap-east-1.amazonaws.com/8f014c5aafb425b2fa414a49eb9ab17269f155937db67a5babe7bc40f9bbae3b/files/1685007975118-DAORAYAKIS3.png"
        ],
        [
            "a",
            "30023:8f014c5aafb425b2fa414a49eb9ab17269f155937db67a5babe7bc40f9bbae3b:aVcTJ7_S0U8QyB1e3bLPD",
            ""
        ],
        [
            "a",
            "30023:8f014c5aafb425b2fa414a49eb9ab17269f155937db67a5babe7bc40f9bbae3b:LlGOjtygN7i4mX0CbLpEo",
            ""
        ]
    ],
    ...other fields
}
```

**Review by monlovesmango — CHANGES_REQUESTED** [review]
hey all, sorry for being MIA. personally I think a new kind is better than using a new 'c' tag for this purpose. we have many more numbers available than letters in the alphabet and using a 'c' tag just for this specific use case seems like a waste. also think it makes sense to have this be a distinct use case from the generalized bookmark lists.

below are my proposed changes for this section. I changed 'excerpt' to 'description' (because excerpt doesn't make much sense when the list is tagging multiple articles) and made the 'd' tag human readable (but it certainly doesn't have to be).

if people like this and it is adopted, the table listing the different kinds of lists will also need to be updated.

**Comment by Darecjo** [comment]
@monlovesmango i think keeping the `kind-30001` is more feasible, since we already implemented the spec within [yakihonne.com](https://yakihonne.com). 

@fiatjaf any thoughts on finalizing the current PR?

**Comment by mostafa-khaldi** [comment]
@fiatjaf How about what @monlovesmango suggested? Seems a good way to gather articles in a special kind.

**Comment by erskingardner** [comment]
- I'm in support of using a new kind over a `"c"` tag. This also helps answer the question for #559 (creating a new list kind focused on a single use case).
- I'm in favor of changing `excerpt` to `description` and `thumbnail` to `image`.

**Comment by fiatjaf** [comment]
I defer to whoever is more directly interested in this, but as a rule of thumb I prefer new kinds for new use cases.

File: 51.md (hunk 8 above)
Comment (vitorpamplona) [reply 51.md:114]: Looks good. 

Although arguably all list types could have the tags below and we could use the `k` tag to be `30023` to allow filtering by blog posts only. 

```json
        [
            "title",
            "Bitcoin Cultural Movement"
        ],
        [
            "description",
            "The advent of Bitcoin has not only revolutionized the financial landscape but has also given rise to a new cultural movement centered around this groundbreaking cryptocurrency. In this era of digital innovation, the concept of curation has emerged as a powerful tool to navigate and make sense of the vast amount of information and ideas surrounding Bitcoin. This article delves into the topic of Bitcoin curation, exploring its purpose, key players involved, and presenting five curated articles that shed light on different aspects of this phenomenon."
        ],
        [
            "thumbnail",
            "https://daorayaki-fs-bucket.s3.ap-east-1.amazonaws.com/8f014c5aafb425b2fa414a49eb9ab17269f155937db67a5babe7bc40f9bbae3b/files/1685007975118-DAORAYAKIS3.png"
        ],
```

Comment (erskingardner) [reply 51.md:114]: +1. I'm actually adding title and description to all lists in the new Listr rebuild.

Comment (monlovesmango) [reply 51.md:114]: 'k' tag is a good idea. I still prefer a dedicated kind. not for technical reasons, just think its more consistent and straightforward for comprehension. also makes filtering easier/cleaner. 

but I will not be the main user of this feature so its really up to those who are invested in this list type whether they want a new dedicated kind or to just use existing list kind and add a 'k' tag.

**Comment by Darecjo** [comment]
L

File: 51.md (hunk 8 above)
Comment (Darecjo) [reply 51.md:114]: L

**Comment by mostafa-khaldi** [comment]
@verbiricha I'm sorry for the late reply, I think this is the best solution, I have already requested before to make curated articles as an independent kind, but I got many replies suggesting otherwise, but here we are agreeing on this again, i'd go for this too

File: 51.md (hunk 8 above)
Comment (Darecjo) [reply 51.md:114]: @mostafa-khaldi L

**Comment by mostafa-khaldi** [comment]
@verbiricha  I'm sorry for the late reply, I think this is the best solution, I have already requested before to make curated articles as an independent kind, but I got many replies suggesting otherwise, but here we are agreeing on this again, i'd go for this too (this is a reply to [your comment](https://github.com/nostr-protocol/nips/pull/492#discussion_r1314887637), mine was put in a holding state for an unknown reason)

**Comment by fiatjaf** [comment]
Yakihonne is doing some curation stuff, right? Has anyone from Yakihonne seem this?

**Comment by verbiricha** [comment]
I was under the impression that @mostafa-khaldi is from YakiHonne project since he linked a YakiHonne video. If not we should reach out for sure.

**Comment by mostafa-khaldi** [comment]
@fiatjaf Yeh, i'm talking on behalf of Yakihonne, I am part of the team, their web developer lol

**Comment by mostafa-khaldi** [comment]
@verbiricha I am indeed

**Comment by Darecjo** [comment]
@fiatjaf the pull request was dedicated to YakiHonne from the start we are behind it.

**Comment by monlovesmango** [comment]
@mostafa-khaldi if you agree with the changes I suggested [here](https://github.com/nostr-protocol/nips/pull/492#pullrequestreview-1531774064) would you mind updating the pr to reflect them?

**Review by mostafa-khaldi — COMMENTED** [review]
@monlovesmango  it works for me

**Comment by mostafa-khaldi** [comment]
@fiatjaf We're gonna close this PR due to the new `kind:30004` and we're gonna comply with its new standards, please confirm the new changes so we can close this PR

**Pull request closed without merging**